project_setup_best_practices.md

```diff
@@ -0,0 +1,165 @@
+[keras-cv](https://github.com/keras-team/keras-cv),
+[keras-nlp](https://github.com/keras-team/keras-nlp),
+and maybe more in the future. It covers linting, formating, testing, continuous
+integration, issues and pull requests tagging, and so on.
+
+The goal of this document is to:
+* Improve the overall quality of the projects. The fact that projects all
+  follow the same standard for dev process, which may evolve through time, will
+  ensure the quality from all aspects.
+* Unify the external contributing experience. The external open-source
+  contributors may contribute to multiple Keras projects by submitting issues
+  or pull requests. They don't need to learn from different contributing
+  guides.
+* Save time for the project leads. They save time by copying and pasting the
+  same setup and by avoiding the listed caveats.
+
+## Testing
+
+### Testing framework
+
+We use [pytest](https://docs.pytest.org/en/6.2.x/) for writing tests for the
+projects, which is the most widely used testing framework for Python in the OSS
+world. The configuration of pytest is
+[here](https://github.com/keras-team/keras-tuner/blob/1.1.0/setup.cfg#L4-L16).
+
+### File locations for the tests
+
+Unit tests should be contained in sibling files, relative to the class or
+utility files they are testing. The name of a test file should follow the
+pattern of `*_test.py`. For example, the tests for
+`/keras_tuner/engine/hyperparameters.py` are in
+`/keras_tuner/engine/hyperparameters_tests.py`.
+
+Integration tests may be contained in their own `/integration` directory, as
+they may require extra files such as data.
+
+While our unit test placement is not suggested in the
+[good practices of pytest](https://docs.pytest.org/en/6.2.x/goodpractices.html)
+doc, we recommend this approach to improve the discoverability of the unit
+tests for new contributors. This discoverability doubles up as a method of
+documentation; when users want to see what `util.utility_function()` does, they
+can simply open the conveniently located sibling file, `util_test.py`.
+
+### Test Coverage
+
+We use [CodeCov](https://about.codecov.io/) to track the test coverage.You may
+also refer to
+[these settings](https://github.com/keras-team/keras-tuner/blob/1.1.0/setup.cfg#L24-L28)
+in `setup.cfg`. We will see more about it in the continuous integration section.
+
+Pytest CodeCov supports a wildcard exclude field, which should be set to
+include `*_test.py`, as to ensure that tests are not included in the code
+coverage count.
+
+### Useful code snippets
+Fix the random seed for all tests:
+[Link1](https://github.com/keras-team/keras-tuner/blob/1.1.0/tests/conftest.py#L8-L17),
+[Link2](https://github.com/keras-team/keras-tuner/blob/master/tests/unit_tests/randomness_test.py),
+[Link3](https://www.tensorflow.org/api_docs/python/tf/keras/utils/set_random_seed).
+
+Create a temporary path for testing: [Link](https://docs.pytest.org/en/6.2.x/tmpdir.html).
+
+## Linting and formatting
+
+We use
```

Comment:
The behavior of these tools could really change on the installed version. It is why is suggest to enforce the versioning with `pre-commit install` and configuration like at https://github.com/keras-team/keras-cv/pull/14.
I suggest to use the docker developer image and `pre-commit` also in  Githun Action CI so that we are sure we are always on the same page with the developers when they validate the in-progress PR code on their own env